syntax = "proto3" ;

service UserManagement {
    rpc create (User) returns (Response);
    rpc read (Id) returns (UserDetails); 
    rpc update_password (UpdatePassword) returns (Response);
    rpc delete (DeleteUser) returns (DeleteResponse);

}

message UpdatePassword {
    Id user_id = 1;
    string old_password = 2;
    string new_password = 3;
}

message DeleteUser {
    Id user_id = 1;
    bool confirm_delete = 2;
}

message Id {
    int32 id = 1;
}

message UserDetails {
    string username = 1;
    string email = 2;
}

message User {
    string username = 1;
    string email = 2;
    string password = 3;
    string confirm_password=4;
}

message Response {
    string message = 1;
}

message DeleteResponse {
    string status = 1;
    string message = 2;
}